File: apps/backend-springboot/src/main/java/br/com/dinheironapalmadamao/backend_springboot/core/registro/provider/BackendRegistro.java
package br.com.dinheironapalmadamao.backend_springboot.core.registro.provider;

import br.com.dinheironapalmadamao.backend_springboot.core.registro.model.Registro;
import br.com.dinheironapalmadamao.backend_springboot.core.registro.model.RegistroDTO;

public interface BackendRegistro {
  public void criar(RegistroDTO registro);

  public Registro obterRegistroPorId(String registro_id);

  public void deletar(String registro_id);

  public void editar(Registro registro);

  public Registro[] obterRegistros();
}

File: packages/registro/src/provider/BackendRegistro.ts
import { Registro, RegistroDTO } from "../model";

export interface BackendRegistro {
  criar(registro: RegistroDTO): Promise<Response | void>;
  obterRegistroPorId(registro_id: string): Promise<Registro | null>;
  deletar(registro_id: string): Promise<void>;
  editar(registro: Registro): Promise<Response | void>;
  obterRegistros(): Promise<Registro[] | null>;
  obterRegistrosPorStatus(status: string): Promise<Registro[] | null>;
}
